### PR TITLE
INTMDB-263: Added Private_link_endpoint_adl resource to go-client

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

Added `PrivateLinkEndpointADL` resource to go-client to manage DataLake and Online Archive private link endpoints

Link to any related issue(s): [INTMDB-263](https://jira.mongodb.org/browse/INTMDB-263)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/privatelink_endpoints_adl.go [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+)
+
+const (
+	privateLinkEndpointsADLPath = "api/atlas/v1.0/groups/%s/privateNetworkSettings/endpointIds"
+)
+
+// PrivateLinkEndpointsADLService is an interface for interfacing with the Private Link Endpoints ADL
+// of the MongoDB Atlas API datalake and online archive.
+//
+// See more: https://docs.atlas.mongodb.com/reference/api/private-endpoints/
```
Comment (gssbzn) [inline mongodbatlas/privatelink_endpoints_adl.go:30]: this link is not correct which leads me to a major issue here.

Given this statement about the go client lib README

> The services of a client divide the API into logical chunks and correspond to
the structure of the Atlas API documentation at
https://docs.atlas.mongodb.com/api/.

You have two options

Add the methods to `OnlineArchiveService` (https://docs.atlas.mongodb.com/reference/api/online-archive/)
Or add the methods to `DataLakeService` (https://docs.mongodb.com/datalake/reference/api/datalakes-api/)

Even adding to both may be the correct thing

Comment (abner-dou) [reply mongodbatlas/privatelink_endpoints_adl.go:30]: my bad, I copy from private_link but I forgot to change this

Comment (abner-dou) [reply mongodbatlas/privatelink_endpoints_adl.go]: already changed

Comment (gssbzn) [reply mongodbatlas/privatelink_endpoints_adl.go]: Thanks for updating the link but that doesn't solve my main comment, let's not add a new service, let's add the method to `OnlineArchiveService`, `DataLakeService` or both

Comment (abner-dou) [reply mongodbatlas/privatelink_endpoints_adl.go]: Oh I understand, but now my concern is, it is okay to use `DataLakeService` when you are using `OnlineArchiveService` and want to create a private_link?

Comment (abner-dou) [reply mongodbatlas/privatelink_endpoints_adl.go]: What happen if the API documentation for the `private_link_endpoint` for this gets a new section? you will need to "refactor" this services?

Comment (gssbzn) [reply mongodbatlas/privatelink_endpoints_adl.go]: > is it okay to use DataLakeService when you are using OnlineArchiveService and want to create a private_link?

I say yes but up to you, remember one of the options is to add to both 

> What happen if the API documentation for the private_link_endpoint for this gets a new section?

We have a "no breaking changes" approach on the client unless strictly necessary, if the docs are moved around we should not move code around, we'll live with the choices made with what we knew today

Comment (abner-dou) [reply mongodbatlas/privatelink_endpoints_adl.go]: Ready! I added this new methods on both datalake and online archive services

**Review by gssbzn — APPROVED** [review]
LGTM

**Comment by abner-dou** [comment]
@gssbzn I cannot merge due my access level, can you merge it, please?

**Pull request merged**